packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -5416,6 +5416,41 @@ void main() {
     expect(state.currentTextEditingValue.text, '12345');
     expect(state.currentTextEditingValue.composing, TextRange.empty);
   });
+
+  testWidgets('Apply formatters for texts added via TextEditingController', (WidgetTester tester) async {
+    // This formatter just upperCase the text.
+    final TextInputFormatter formatter = TextInputFormatter.withFunction((oldValue, newValue) {
+      return newValue.copyWith(text: newValue.text.toUpperCase());
+    });
+
+    final Widget widget = MaterialApp(
+      home: EditableText(
+        backgroundCursorColor: Colors.grey,
+        controller: controller,
+        focusNode: focusNode,
+        inputFormatters: <TextInputFormatter>[formatter],
+        style: textStyle,
+        cursorColor: cursorColor,
+        selectionControls: materialTextSelectionControls,
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+
+    final EditableTextState state = tester.state<EditableTextState>(find.byType(EditableText));
+
+    // This was supposed to rebuild the widget and call _formatAndSetValue
+    // controller.text = 'flutter';
+    // Since the line above does not update the widget the line bellow
+    // is a workaround simulating changes from controller.text.
+    state.updateEditingValue(const TextEditingValue(text: 'flutter'));
```

Comment:
@justinmc FYI this line is a work around for the controller.text, because it is not updating the widget.

Reply:
May be because of this TODO
https://github.com/flutter/flutter/blob/0205911115facdd70f712b37aeff4ba43bbcd291/packages/flutter/lib/src/widgets/editable_text.dart#L2230-L2238